pkg/restore/crd.go

```diff
@@ -0,0 +1,225 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var getCrName = `
+	kubectl -n %s get %s -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.deletionGracePeriodSeconds}{"\n"}{end}' | awk '$2=="0" {print $1}'
+`
+
+var getCrYamlContent = `
+	kubectl -n %s get %s %s -oyaml > %s.yaml
+`
+
+var getCrUID = `
+	kubectl -n %s get %s %s -o 'jsonpath={.metadata.uid}'
+`
+
+var removeFinalizers = `
+	kubectl -n %s patch %s/%s --type json --patch='[ { "op": "remove", "path": "/metadata/finalizers" } ]'
+`
+
+var recreateCr = `
+	kubectl create -f %s
+`
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	crd := args[0]
+	command := fmt.Sprintf(getCrName, clusterNamespace, crd)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
+
+	if len(args) == 2 {
+		crName = args[1]
+	}
+
+	logging.Info("Restoring crd %s name %s", crd, crName)
+
+	logging.Info("Scaling down the operator to 0")
+	err := debug.SetDeploymentScale(ctx, k8sclientset.Kube, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		logging.Fatal(err)
+	}
+
+	logging.Info("Backing up kubernetes and crd resources")
+	command = fmt.Sprintf(getCrYamlContent, clusterNamespace, crd, crName, crd)
+	exec.ExecuteBashCommand(command)
+	logging.Info("Backed up crd %s named %s in file %s.yaml", crd, crName, crd)
+
+	webhookConfigName := "rook-ceph-webhook"
+	logging.Info("deleting validating webhook %s if present", webhookConfigName)
+	err = k8sclientset.Kube.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(ctx, webhookConfigName, v1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		logging.Fatal(fmt.Errorf("failed to delete validating webhook %s. %v", webhookConfigName, err))
+	}
+
+	logging.Info("Fetching %s %s uid", crd, crName)
+	command = fmt.Sprintf(getCrUID, clusterNamespace, crd, crName)
+	uid := exec.ExecuteBashCommand(command)
+	logging.Info("Successfully fetched uid %s from %s %s", uid, crd, crName)
+
+	removeOwnerRefOfUID(ctx, k8sclientset, operatorNamespace, clusterNamespace, uid)
+
+	logging.Info("removing finalizers from %s %s", crd, crName)
```

Comment:
nit: Some of these log statements start with capital letter and some with lower case. How about starting with upper case for consistency?
```suggestion
	logging.Info("Removing finalizers from %s %s", crd, crName)
```